### PR TITLE
1002 Enhance DecathlonDataset to load properties

### LINKED ISSUE #1002: Add more properties in DecathlonDataset
**Is your feature request related to a problem? Please describe.**
We can load properties from the JSON config file of DecathlonDataset, for example: number of classes.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #1002 .

### Description
This PR added support to load expected properties from the JSON config.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/apps/datasets.py [hunk 1]
```diff
@@ -241,11 +245,28 @@ def __init__(
                 f"Cannot find dataset directory: {dataset_dir}, please use download=True to download it."
             )
         data = self._generate_data_list(dataset_dir)
+        if property_keys is not None:
+            self._properties = load_decathlon_properties(os.path.join(dataset_dir, "dataset.json"), property_keys)
```
Comment (wyli) [inline monai/apps/datasets.py:249]: I think we should always load all the properties, this is only a very small amount of data and keep the function signature simple

File: monai/apps/datasets.py [hunk 2]
```diff
@@ -241,11 +245,28 @@ def __init__(
...
+        Get the loaded properties of dataset with specified keys.
+        If no keys specified, return all the loaded properties.
+
+        """
+        if keys is None:
+            return self._properties
+        elif self._properties is not None:
+            return {key: self._properties[key] for key in ensure_tuple(keys)}
+        else:
+            return None
```
Comment (wyli) [inline monai/apps/datasets.py:268]: should return `{}` here?

File: monai/apps/datasets.py [hunk 3]
```diff
@@ -241,11 +245,28 @@ def __init__(
...
+    def get_properties(self, keys: Optional[Union[Sequence[str], str]] = None):
+        """
+        Get the loaded properties of dataset with specified keys.
+        If no keys specified, return all the loaded properties.
+
+        """
+        if keys is None:
+            return self._properties
+        elif self._properties is not None:
+            return {key: self._properties[key] for key in ensure_tuple(keys)}
```
Comment (wyli) [inline monai/apps/datasets.py:266]: what if `key not in self.properties`?

**Review by wyli — COMMENTED** [review]
thanks, I put some comments inline

File: monai/apps/datasets.py (hunk 2 above)
Comment (Nic-Ma) [reply monai/apps/datasets.py:268]: updated in the latest commit.

File: monai/apps/datasets.py (hunk 3 above)
Comment (Nic-Ma) [reply monai/apps/datasets.py:266]: I think it will raise an exception by dict object directly?
Thanks.

File: monai/apps/datasets.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/apps/datasets.py:249]: OK, updated in the latest commit.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for your review.
I updated the PR according to your comments.
Could you please help review it again?

Thanks.

**Review by wyli — APPROVED** [review]

**Pull request merged**